Overhaul color picker module

dterrahe opened the pull request:
As promised here https://github.com/darktable-org/darktable/issues/6282#issuecomment-692161038

![image](https://user-images.githubusercontent.com/1549490/93878807-93218280-fcd2-11ea-92ed-9a6ab8b1a037.png)

Removed broken functionality and streamlined module.
Improved the tooltip.
Works better with narrow panels.

Fixes #6282

TurboGit commented on src/libs/colorpicker.c:
```diff
@@ -466,194 +452,165 @@ static void _set_sample_area(dt_lib_module_t *self, float size)
+  gtk_widget_set_valign(data->color_mode_selector, GTK_ALIGN_END);
+  gtk_box_pack_start(GTK_BOX(picker_row), data->color_mode_selector, TRUE, TRUE, 0);
+
+  gtk_box_pack_start(GTK_BOX(picker_row), gtk_label_new("  "), FALSE, FALSE, 0);
```
This should be done in CSS if needed.

dterrahe replied:
removed

TurboGit commented:
The line with statistic & model is too crowded and in some language (here French) the label are ellipsised.

![image](https://user-images.githubusercontent.com/467069/93911254-7604a800-fd02-11ea-81f2-539dbbe13e77.png)

I think the statistic & model should be on 2 lines or the label "model" removed as Lab RGB is clear on this widget?

dterrahe commented:
Mostly I added the label, because without it, the leftmost combobox kindof floats in the air, because it is right aligned; "mean" doesn't start at the left border. I'll experiment a little with other layouts (in French). Removing "model" might not help, because it won't change the split allocated to the left combobox, I think.

dterrahe commented:
I'm just seeing #6352 pop up which overlaps significantly with this PR.

Let me know what you want to do with that. If you have the intention of merging it, it would make sense to do that first and wait with this one until I have rebased it incorporating the needed changes, rather than forcing msdm to deal with that.

Nilvus commented:
> Mostly I added the label, because without it, the leftmost combobox kindof floats in the air, because it is right aligned; "mean" doesn't start at the left border. I'll experiment a little with other layouts (in French). Removing "model" might not help, because it won't change the split allocated to the left combobox, I think.

Why not simply move picker under that line and then add a zone button. That would aerate text there and have a visual way to understand both possibilities: point and zone picker. I have understand that actually zone could be set with ctrl+click (tooltip missing) but could be simpler to have two buttons and so a more aerate UI, by remaining a better one than actual one.

dterrahe commented:
Removed the combobox labels, so they don't get ellipsized anymore.
Moved names to end of line in tooltip, which causes less problems with non-utf8 languages (Russian now looking fine).
Added button tooltip.

![image](https://user-images.githubusercontent.com/1549490/93935128-10251a00-fd1c-11ea-81f6-851714da52f2.png)

@Nilvus We don't have separate icons for point and area pickers and using ctrl is more consistent with elsewhere, but technically your suggestion would be possible. Especially after removing the labels, I don't feel aerating the text is necessary. Rather the opposite; I added the labels because in English the dialog is quite sparse.

TurboGit commented:
@dterrahe : I have already reviewed this PR, so #6352 will have to be reworked on top of this. It is rare to see such conflicts, but that's good news as it means quite a lot of changes in dt :)

TurboGit commented:
Ok, works for me. Really nice and functional. I really like the large color patch, really handy to visually evaluate the color. All goof now, merging, we'll have more field testing. Thanks.
